src/note_compiler.c

```diff
@@ -53,27 +55,50 @@ static int finish_note(NoteCompiler* compiler, Note* note) {
 		compiler->finished = true;
 	}
 
-	if(!compiler->finished && !rest) {
+	if(!rest) {
 
-		switch(c) {
+		if(!compiler->finished) {
 
-			case ' ':
-			case '\t':
-				note_finished = true;
-				break;
+			switch(c) {
 
-			case '#':
-				advance(compiler);
-				half++;
-				break;
+				case ' ':
+				case '\t':
+					note_finished = true;
+					break;
 
-			case 'b':
-				advance(compiler);
-				half--;
-				break;
+				case '#':
+					advance(compiler);
+					half++;
+					break;
 
-			default:
-				break;
+				case 'b':
+					advance(compiler);
+					half--;
+					break;
+
+				default:
+					break;
+			}
+		}
+
+		while(!frequence_parsing_finished && !compiler->finished && (c = peek(compiler)) >= 0) {
```

Comment:
We're reaching a state where we might consider refactoring the `NoteCompiler` and making more use of internal states, as for the lexer and interpreter.